lib/sycamore/sycamore/transforms/merge_elements.py

```diff
@@ -611,6 +611,195 @@ def postprocess_element(self, elem: Element) -> Element:
+            if element2.text_representation:
+                if element2.data["_header"]:
+                    element2.text_representation = element2.data["_header"] + "\n" + element2.text_representation
+            else:
+                element2.text_representation = element2.data["_header"]
+        return False
+
+    def merge(self, elt1: Element, elt2: Element) -> Element:
+        """Merge two elements; the new element's fields will be set as:
+            - type: "Section-header" , "MergedText"
+            - binary_representation: elt1.binary_representation + elt2.binary_representation
+            - text_representation: elt1.text_representation + elt2.text_representation
+            - bbox: the minimal bbox that contains both elt1's and elt2's bboxes
+            - properties: elt1's properties + any of elt2's properties that are not in elt1
+            note: if elt1 and elt2 have different values for the same property, we take elt1's value
+            note: if any input field is None we take the other element's field without merge logic
+
+        Args:
+            element1 (Tuple[Element, int]): the first element (and number of tokens in it)
+            element2 (Tuple[Element, int]): the second element (and number of tokens in it)
+
+        Returns:
+            Tuple[Element, int]: a new merged element from the inputs (and number of tokens in it)
+        """
+
+        tok1 = elt1.data["token_count"]
+        tok2 = elt2.data["token_count"]
+        new_elt = Element()
+
+        if elt1.type in ["Section-header", "Title"] and elt2.type in ["Section-header", "Title"]:
+            new_elt.type = "Section-header"
+        else:
+            new_elt.type = "Text"
+
+        # Merge binary representations by concatenation
+        if elt1.binary_representation is None or elt2.binary_representation is None:
+            new_elt.binary_representation = elt1.binary_representation or elt2.binary_representation
+        else:
+            new_elt.binary_representation = elt1.binary_representation + elt2.binary_representation
+
+        # Merge text representations by concatenation with a newline
+
+        new_elt_text_representation = "\n".join(filter(None, [elt1.text_representation, elt2.text_representation]))
+        new_elt.text_representation = new_elt_text_representation if new_elt_text_representation else None
+        # new_elt.text_representation = elt1.text_representation + "\n" + elt2.text_representation
+        new_elt.data["token_count"] = tok1 + 1 + tok2
+
+        # Merge bbox by taking the coords that make the largest box
+        if elt1.bbox is None and elt2.bbox is None:
+            pass
+        elif elt1.bbox is None or elt2.bbox is None:
+            new_elt.bbox = elt1.bbox or elt2.bbox
+        else:
+            new_elt.bbox = BoundingBox(
+                elt1.bbox.x1,
+                elt1.bbox.y1,
+                elt1.bbox.x2,
+                elt1.bbox.y2,
+            )
```

Comment:
Don't we want to use the `min` and `max` logic here? Like below:
```python
new_elt.bbox = BoundingBox(
                min(elt1.bbox.x1, elt2.bbox.x1),
                min(elt1.bbox.y1, elt2.bbox.y1),
                max(elt1.bbox.x2, elt2.bbox.x2),
                max(elt1.bbox.y2, elt2.bbox.y2),
            )
```

Reply:
I am unsure, what if it is extending multiple pages? We never made a decision about how the UI would look like (do we even show the chunked document?